File: grokking-algo/practice/sele.py

arr = [10, 1, 2, 4, 7, 9, 0]

def selection_sort(arr: list) -> list:
    for i in range(len(arr)):
        min_index = i
        for j in range(i+1, len(arr)):
            if arr[min_index] > arr[j]:
                min_index = j
        
        arr[i], arr[min_index] = arr[min_index], arr[i]
    
    return arr


print(selection_sort(arr))

File: grokking-algo/practice/sele.js


/**
 * 
 * @param {Array} array 
 */
function SelectionSort(array) {
  for (let i = 0; i < array.length; ++i) {
    let min_index = i;
    for (let j = i + 2; j < array.length; ++j){
      if (array[min_index] > array[j]) {
        min_index = j;
      }
    }
    [array[i], array[min_index]] = [array[min_index], array[i]];
  }
  return array;  
}

let L = [3, 1, 41, 59, 26, 53, 59];
console.log(SelectionSort(L))